File: src/main/java/org/jhapy/cqrs/command/SubmitUploadCommand.java
package org.jhapy.cqrs.command;

import lombok.*;
import org.axonframework.modelling.command.TargetAggregateIdentifier;

import javax.validation.constraints.NotNull;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class SubmitUploadCommand extends AbstractBaseCommand {
  @TargetAggregateIdentifier private UUID id = UUID.randomUUID();

  @NotNull private UUID uploadId;

  @NotNull private String filename;
}

File: src/main/java/org/jhapy/cqrs/command/SubmitUploadCommand.kt
package org.jhapy.cqrs.command

import org.axonframework.modelling.command.TargetAggregateIdentifier
import java.util.*
import javax.validation.constraints.NotNull

class SubmitUploadCommand(
    val uploadId: @NotNull UUID,
    val filename: @NotNull String
) : AbstractBaseCommand() {
    @TargetAggregateIdentifier
    val id: UUID = UUID.randomUUID()
}